Harlowe: Storylets

videlais opened the pull request.

greyelf commented on pages/storylets/harlowe/harlowe_storylets.md:
```diff
@@ -0,0 +1,78 @@
+The metaphor of a deck of cards is often useful in understanding storylets. A card is able to be drawn from a deck if its requirements are met. Through defining many cards (passages in Twine) and conditions under which they can be used (their requirements), dynamic storytelling can take place as different arrangements become possible through writing simple rules for each card.
+
+The [`(storylet:)` macro](https://twine2.neocities.org/#macro_storylet) defines a passage as a storylet in Harlowe. It should appear as either the first line or as near to the top of the passage as possible. To define if it is available (called "open" in Harlowe) or not, it uses a [`when` lambda](https://twine2.neocities.org/#type_lambda) for determining, in the future, if the requirements are me ...
+
```
> To define if it is available (called "open" in Harlowe) or not, it uses a `when` lambda for determining, in the future, if the requirements are met using story and temporary variables.

While this statement is factually true, and it includes two important terms ('open' and 'when') related to storylets, I wonder if it is too technical in wording for those with less programming experience and if not enough emphasise is given to the key 'open' term?

The following is not a replacement suggestion, just an example of one way to add emphasise to the 'open' term.
> A Harlowe storylet is considered 'open' when its lamba conditional expression evaluates to true.

ChapelR replied:
If you're concerned about the 'open' language, maybe try something like 
> A Harlowe storylet is considered **open**, meaning it is available to be "drawn," when its lamba conditional expression evaluates to true.

ChapelR commented:
> The (link-storylet:) macro selects the first open storylet based on a number passed to the macro.  

Some potential confusion here&mdash;I think changing the phrasing to something like this may be clearer.
> The (link-storylet:) macro selects the first open storylet, or, if a number passed to the macro, the storylet associated with that number.  

I think the important part is there needs to be an *or* of some kind, to clarify that it selects the first open storylet only when a number isn't specified.

ChapelR commented on pages/storylets/harlowe/harlowe_storylets.md:
```diff
@@ -0,0 +1,78 @@
+
+The [`(storylet:)` macro](https://twine2.neocities.org/#macro_storylet) defines a passage as a storylet in Harlowe. It should appear as either the first line or as near to the top of the passage as possible. To define if it is available (called "open" in Harlowe) or not, it uses a [`when` lambda](https://twine2.neocities.org/#type_lambda) for determining, in the future, if the requirements are me ...
+
+The [`(link-storylet:)` macro](https://twine2.neocities.org/#macro_link-storylet) selects the first open storylet based on a number passed to the macro. To access the fourth open storylet, for example, `(link-storylet: 4)` could be used. If no storylets are open, a third value can be used in the format of `(link-storylet: 4, "No storylets")` where the fourth open storylet would be used or the val ...
```
Suggest changing the wording here as I feel it's a little confusing. Something like:
> The (link-storylet:) macro selects the first open storylet, or, if a number passed to the macro, the storylet associated with that number.

I think it needs clarified a bit is all.

videlais replied:
> Suggest changing the wording here as I feel it's a little confusing. Something like:
> 
> > The (link-storylet:) macro selects the first open storylet, or, if a number passed to the macro, the storylet associated with that number.
> 
> I think it needs clarified a bit is all.

I thought that was the case, too, but it does not work that way, it seems.

Using `(link-storylet:)` by itself produces the following error:

> the (link-storylet:) macro must only be given a whole number or a string or a "where ..." lambda, (optional) a whole number or a string or a "where ..." lambda and (optional) a string, in that order.

ChapelR commented on pages/storylets/harlowe/harlowe_storylets_twee.txt:
```diff
@@ -0,0 +1,53 @@
+[[Change search?->Start]]
+
+
+:: Matches {"position":"198,299","size":"100,100"}
```
I feel like the two storylets should be together at the bottom and the `Matches` passage should be right after `Start`. Not a big deal though.

ChapelR reviewed (approved):
I left two minor suggestions I wouldn't mind seeing addressed, but they aren't major issues.

videlais commented:
Updates:

- Revised storylet language around open and lambda types.
- Updated `(link-storylet:)` language. Current version requires either a number or `when` lambda.
- Switched passage ordering in Twee code.
